src/main/java/com/redhat/devtools/lsp4ij/settings/jsonSchema/LSPServerConfigurationJsonSchemaManager.java

```diff
@@ -0,0 +1,50 @@
+package com.redhat.devtools.lsp4ij.settings.jsonSchema;
+
+import com.intellij.openapi.project.Project;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class LSPServerConfigurationJsonSchemaManager {
+
+    public static LSPServerConfigurationJsonSchemaManager getInstance(@NotNull Project project) {
+        return project.getService(LSPServerConfigurationJsonSchemaManager.class);
+    }
+
+    private final List<LSPServerConfigurationJsonSchemaFileProvider> providers;
+
+    public LSPServerConfigurationJsonSchemaManager(@NotNull Project project) {
+        providers = new ArrayList<>();
+        for (int i = 0; i < 100; i++) {

```

Comment:
I'm assuming you're just adding a bank of these large enough to accommodate any foreseeable use case? If that's the case, migth be good to extract `100` as a constant and document the assumptions being made here. Or is there a way to make this more dynamic based on actual usage patterns?